docs/src/pages/migration-guide.md

```diff
@@ -0,0 +1,225 @@
+---
+layout: ~/layouts/MainLayout.astro
+title: Migrating to v0.21
+description: How to migrate projects from Astro v0.20.
+---
+
+Starting in v0.21, Astro is now built with [Vite].
+As a result, configurations written in `snowpack.config.mjs` should be moved into `astro.config.mjs`.
+
+```js
+// @ts-check
+
+/** @type {import('astro').AstroUserConfig} */ 
+export default ({
+  renderers: [],
+  vite: {
+    plugins: []
+  }
+})
+```
+
+To learn more about configuring Vite, please visit their [configuration guide](https://vitejs.dev/config/).
+
+
+
+## Aliases
+
+In Astro v0.21, import aliases can be added using `tsconfig.json` or `jsconfig.json`.
+
+```json
+{
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {
+      "@/components/*": ["components/*"]
+    }
+  }
+}
+```
+
+These aliases are integrated automatically into [VSCode](https://code.visualstudio.com/docs/languages/jsconfig) and other editors.
+
+
+
+## Aliases (#1749)
+
+In Astro v0.21, import aliases can also be added using the `alias` property.
+
+```js
+export default {
+  vite: {
+    resolve: {
+      alias: {
+        @components: '/src/components',
+      }
+    }
+  }
+}
+```
+
+Note: Always start alias values with `/`. Relative alias starting with `.` will be used as-is and will not be resolved.
+
+
+
+## Passing variables into scripts and styles
+
+In Astro v0.21, variables can be passed from Frontmatter into inline `<style>` and `<script>`.
+
+```astro
+---
+// tick.astro
+const colors = { foregroundColor: "rgb(221 243 228)", backgroundColor: "rgb(24 121 78)" }
+---
+<style define:vars={green}>
```

Comment:
`define:vars={colors}` right?

Reply:
Thanks! Fixed in f57616e29963b9ce4598b37451ca6e1fbbe66176